documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/migration/ServletToHandlerDocs.java

```diff
@@ -0,0 +1,643 @@
+            //   - servletRequest.getMethod();
+            String method = request.getMethod();
+
+            // Gets the request protocol name and version.
+            // Replaces:
+            //   - servletRequest.getProtocol();
+            String protocol = request.getConnectionMetaData().getProtocol();
+
+            // Gets the full request URI.
+            // Replaces:
+            //   - servletRequest.getRequestURL();
+            String fullRequestURI = request.getHttpURI().asString();
+
+            // Gets the request context.
+            // Replaces:
+            //   - servletRequest.getServletContext()
+            Context context = request.getContext();
+
+            // Gets the context path.
+            // Replaces:
+            //   - servletRequest.getContextPath()
+            String contextPath = context.getContextPath();
+
+            // Gets the request path.
+            // Replaces:
+            //   - servletRequest.getRequestURI();
+            String requestPath = request.getHttpURI().getPath();
+
+            // Gets the request path after the context path.
+            // Replaces:
+            //   - servletRequest.getServletPath() + servletRequest.getPathInfo()
+            String pathInContext = Request.getPathInContext(request);
+
+            // Gets the request query.
+            // Replaces:
+            //   - servletRequest.getQueryString()
+            String queryString = request.getHttpURI().getQuery();
+
+            // Gets request parameters.
+            // Replaces:
+            //   - servletRequest.getParameterNames();
+            //   - servletRequest.getParameter(name);
+            //   - servletRequest.getParameterValues(name);
+            //   - servletRequest.getParameterMap();
+            Fields queryParameters = Request.extractQueryParameters(request, UTF_8);
+            Fields allParameters = Request.getParameters(request);
+
+            // Gets cookies.
+            // Replaces:
+            //   - servletRequest.getCookies();
+            List<HttpCookie> cookies = Request.getCookies(request);
+
+            // Gets request HTTP headers.
+            // Replaces:
+            //   - servletRequest.getHeaderNames()
+            //   - servletRequest.getHeader(name)
+            //   - servletRequest.getHeaders(name)
+            //   - servletRequest.getDateHeader(name)
+            //   - servletRequest.getIntHeader(name)
+            HttpFields requestHeaders = request.getHeaders();
+
+            // Gets the request Content-Type.
+            // Replaces:
+            //   - servletRequest.getContentType()
+            String contentType = request.getHeaders().get(HttpHeader.CONTENT_TYPE);
+
+            // Gets the request Content-Length.
+            // Replaces:
+            //   - servletRequest.getContentLength()
+            //   - servletRequest.getContentLengthLong()
+            long contentLength = request.getHeaders().getLongField(HttpHeader.CONTENT_LENGTH);
```

Comment:
Note that we do have `request.getLength()`, which goes to the `MetaData.Request#getContentLength`, which I think will always have the header length if available.
Will it ever have a length even if there is no header?

Reply:
We should check why we have code getting the header directly

Reply:
> Fields allParameters = Request.getParameters(request);

I don't understand the difference `allParameters` and `queryParameters`.

Reply:
Query parameters are just from the query string. All parameters include any parameters read from the request body

Reply:
When you say parameters read from the request body, I assume you mean HTTP POST form parameters? This isn't necessarily obvious from the documentation.